pkg/agent/openflow/pipeline.go

```diff
@@ -586,26 +592,77 @@ func (f *featurePodConnectivity) podClassifierFlow(podOFPort uint32, isAntreaFle
 
 // podUplinkClassifierFlows generates the flows to mark the packets with target destination MAC address from uplink/bridge
 // port, which are needed when uplink is connected to OVS bridge and Antrea IPAM is configured.
-func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr) []binding.Flow {
+func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr, vlanID uint16) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
-	return []binding.Flow{
+	var flows []binding.Flow
+	ctZoneTypeRegMark := IPCtZoneTypeRegMark
+	if len(f.ipProtocols) == 1 && f.ipProtocols[0] == binding.ProtocolIPv6 {
+		ctZoneTypeRegMark = IPv6CtZoneTypeRegMark
+	}
+	nonVLAN := true
+	if vlanID > 0 {
+		nonVLAN = false
+	}
+	flows = append(flows,
 		// This generates the flow to mark the packets from uplink port.
-		ClassifierTable.ofTable.BuildFlow(priorityHigh).
+		ClassifierTable.ofTable.BuildFlow(priorityNormal).
 			Cookie(cookieID).
 			MatchInPort(config.UplinkOFPort).
 			MatchDstMAC(dstMAC).
+			MatchVLAN(nonVLAN, vlanID, nil).
+			Action().LoadRegMark(ctZoneTypeRegMark).
+			Action().LoadToRegField(VLANIDField, uint32(vlanID)).
 			Action().LoadRegMark(FromUplinkRegMark).
 			Action().GotoStage(stageConntrackState).
 			Done(),
-		// This generates the flow to mark the packets from bridge local port.
-		ClassifierTable.ofTable.BuildFlow(priorityHigh).
-			Cookie(cookieID).
-			MatchInPort(config.BridgeOFPort).
-			MatchDstMAC(dstMAC).
-			Action().LoadRegMark(FromBridgeRegMark).
-			Action().GotoStage(stageConntrackState).
-			Done(),
+	)
+	if vlanID == 0 {
+		flows = append(flows,
+			// This generates the flow to mark the packets from bridge local port.
+			ClassifierTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
```

Comment:
Use a general function with parameter inPort for the following 3 flows?

Reply:
I tried this method, but it needs many if...else... , thus I think current implementation is OK.